internal/job/job.go

```diff
@@ -0,0 +1,223 @@
+package job
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"sync/atomic"
+
+	"github.com/joshuarubin/teleport-job-worker/internal/config"
+	"github.com/joshuarubin/teleport-job-worker/internal/safebuffer"
+	"github.com/joshuarubin/teleport-job-worker/internal/user"
+)
+
+// Job represents a system process
+type Job struct {
+	cfg    *config.Config
+	id     ID
+	userID user.ID
+	cmd    *exec.Cmd
+	buf    *safebuffer.Buffer
+
+	startOnce sync.Once
+
+	stopOnce sync.Once
+	stopErr  atomic.Value
+
+	status atomic.Uint32
+
+	done   chan struct{}
+	cmdErr error // only safe to read after done has closed
+}
+
+// New creates, but does not start a new job
+func New(
+	ctx context.Context,
+	cfg *config.Config,
+	userID user.ID,
+	command string,
+	args []string,
+	env []string,
+) (*Job, error) {
+	id, err := NewID()
+	if err != nil {
+		return nil, err
+	}
+
+	done := make(chan struct{})
+
+	j := Job{
+		cfg:    cfg,
+		id:     id,
+		userID: userID,
+		done:   done,
+		cmd:    exec.CommandContext(ctx, command, args...),
```

Comment:
Is it a good design to have the job depending on the parent context? If the parent context is the RPC context, won't it be canceled as soon as the job start RPC ends?